src/main/java/arona/Arona.java

```diff
@@ -0,0 +1,123 @@
+package arona;
+
+import java.util.Scanner;
+
+import arona.commands.DeadlineCommand;
+import arona.commands.DeleteCommand;
+import arona.commands.EventCommand;
+import arona.commands.FindCommand;
+import arona.commands.ListCommand;
+import arona.commands.MarkCommand;
+import arona.commands.ToDoCommand;
+import arona.commands.UnMarkCommand;
+import arona.parser.Parser;
+import arona.storage.Storage;
+import arona.task.TaskList;
+import arona.ui.Ui;
+
+/**
+ * Arona is a simple command-line task management application that allows users to manage their tasks.
+ * Users can add tasks, mark tasks as done, delete tasks, and list all tasks.
+ */
+public class Arona {
+    private Storage storage;
+    private TaskList tasks;
+    private Ui ui;
+
+    /**
+     * Initializes a new instance of the Arona application with the specified file path for data storage.
+     *
+     * @param filePath The file path where task data is stored.
+     */
+    public Arona(String filePath) {
+        ui = new Ui();
+        storage = new Storage(filePath);
+        tasks = new TaskList(storage);
+    }
+
+    /**
+     * Runs the Arona application, allowing users to interact with the task management system
+     * via the command-line interface.
+     */
+    public void run() {
+        ui.showWelcomeMessage();
+
+        Scanner scanner = new Scanner(System.in);
+        String userInput;
+
+        while (true) {
+            userInput = scanner.nextLine().toLowerCase().trim();
+
+            if (userInput.isEmpty()) {
+                continue;
+            }
+
+            String[] inputTokens = Parser.parseUserInput(userInput);
+            String command = Parser.getCommand(inputTokens);
+
+            if (command.equals("bye")) {
+                if (inputTokens.length == 1) {
+                    break;
+                }
+            }
+            try {
+                switch (command) {
```

Comment:
I like the way you encapsulate the commands within their own respective classes. 
However, I feel that this switch block should belong in the Parser class since a Parser decides what to do with the given input. In this case, Arona class seems to be the one deciding what to do based on the user input.